test/preconditioner/batch_jacobi_kernels.cpp

```diff
@@ -0,0 +1,231 @@
+
+            for (int r = 0; r < bsize; r++) {
+                for (int c = 0; c < bsize; c++) {
+                    const auto ref_val_ptr =
+                        ref_dense_block_ptr + r * ref_stride + c;
+                    const auto d_val_ptr = d_dense_block_ptr + r * d_stride + c;
+
+                    ValueType val;
+                    exec->get_master()->copy_from(exec.get(), 1, d_val_ptr,
+                                                  &val);
+                    GKO_EXPECT_NEAR(*ref_val_ptr, val, tol);
+                }
+            }
+        }
+    }
+}
+
+
+}  // namespace detail
+
+
+template <typename T>
+class BatchJacobi : public CommonTestFixture {
+protected:
+    using value_type = T;
+    using real_type = gko::remove_complex<value_type>;
+    using Mtx = gko::batch::matrix::Csr<value_type, int>;
+    using BMVec = gko::batch::MultiVector<value_type>;
+    using BJ = gko::batch::preconditioner::Jacobi<value_type>;
+
+    BatchJacobi()
+        : ref_mtx(
+              gko::share(gko::test::generate_diag_dominant_batch_matrix<Mtx>(
+                  ref, nbatch, nrows, false, 4 * nrows - 3))),
+          d_mtx(gko::share(Mtx::create(exec))),
+          ref_b(gko::test::generate_random_batch_matrix<BMVec>(
+              nbatch, nrows, 1, std::uniform_int_distribution<>(nrows, nrows),
+              std::normal_distribution<real_type>(),
+              std::default_random_engine(34), ref)),
+          d_b(BMVec::create(exec,
+                            gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1)))),
+          ref_x(BMVec::create(
+              ref, gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1)))),
+          d_x(BMVec::create(exec,
+                            gko::batch_dim<2>(nbatch, gko::dim<2>(nrows, 1))))
+    {
+        d_mtx->copy_from(ref_mtx.get());
+        d_b->copy_from(ref_b.get());
+        ref_scalar_jacobi_prec =
+            BJ::build().with_max_block_size(1u).on(ref)->generate(ref_mtx);
+        d_scalar_jacobi_prec =
+            BJ::build().with_max_block_size(1u).on(exec)->generate(d_mtx);
+        ref_block_jacobi_prec = BJ::build()
+                                    .with_max_block_size(max_blk_sz)
+                                    .on(ref)
+                                    ->generate(ref_mtx);
+
+        // so that the block pointers are exactly the same for ref and device
+        const int* block_pointers_generated_by_ref =
+            ref_block_jacobi_prec->get_const_block_pointers();
+        const auto num_blocks_generated_by_ref =
+            ref_block_jacobi_prec->get_num_blocks();
+
+        gko::array<int> block_pointers_for_device(
+            this->exec, block_pointers_generated_by_ref,
+            block_pointers_generated_by_ref + num_blocks_generated_by_ref + 1);
+
+        d_block_jacobi_prec =
+            BJ::build()
+                .with_max_block_size(max_blk_sz)
+                .with_block_pointers(block_pointers_for_device)
```

Comment:
why does this use already generated block pointers? Because of that, you don't check if the same block pointers are generated.